lib/IR/LegacyPassManager.cpp

```diff
@@ -685,6 +685,24 @@ void PMTopLevelManager::schedulePass(Pass *P) {
     PP->assignPassManager(activeStack, getTopLevelPassManagerType());
   }
 
+  // HLSL Change - begin
+  if (PI && !PI->isAnalysis() && (this->HLSLPrintBeforeAll || (this->HLSLPrintBefore.size() && this->HLSLPrintBefore.count(PI->getPassArgument())))) {
+    class direct_stderr_stream : public raw_ostream {
```

Comment:
Can `direct_stderr_stream` be shared now between this and print after, rather than copied?

Reply:
Yes it can. Fixed in the latest update (where I also clang-formatted this and the PrintAfter block it was copied from)